### ISSUE TITLE
<Ctrl-O> and :ju after startup causes temporary lockup

### ISSUE THREAD

**Issue by qgates** [neovim/neovim#27974]
### Problem

Running nvim and immediately pressing <Ctrl-O> or issuing `:ju` command causes nvim to freeze for 20+ seconds before the jump to file occurs / jumplist appears. Thereafter, <Ctrl-O>, `:ju` and <Ctrl-I> are responsive.

The problem appears to be resolved if all the filepaths in the jumplist are available. In my case, the jumplist contains references to files like X:\dev\xyz.txt, where drive X: is mapped to a network share that is unavailable some of the time. When files on X: are available, jumplist commands on startup are immediate as expected. When X: is disconnected / offline, the problem occurs.

### Steps to reproduce

.

### Expected behavior

.

### Neovim version (nvim -v)

v0.10.0-dev-2663+gc1c6c1ee1

### Vim (not Nvim) behaves the same?

no

### Operating system/version

Windows 10

### Terminal name/version

Windows Terminal preview

### $TERM environment variable

tmux

### Installation

Nightly win64 build

**Comment by zeertzjq** [comment]
Sounds similar to #12969. Does `:set shada+=rX:` fix the problem?

**Comment by qgates** [comment]
Doesn't appear to. But my jumplist also contains references to UNC paths like \\machine\share\file.txt. So I tried `:set shada+=r\\machine\share` in addition to your suggestion but neither/both have any effect on the delay with `ju` or `<Ctrl-O>`. I think this may be a different issue. When `\\machine\share` is available, which is also the share that X: maps to, there are no delays.

I experience no other issues, eg. delays when quitting nvim and, in case unclear, the shada file is stored on the local (C:) drive.

**Comment by zeertzjq** [comment]
For `:set` backslashes need to be escaped, so it should be `:set shada+=r\\\\machine\\share` actually. Does that work?

**Comment by zeertzjq** [comment]
Or maybe this issue is caused by a plugin trying to access locations in jumplist?

**Comment by qgates** [comment]
Thanks for the escaping tip, that escaped my attention (no pun intended). So I added the following to `init.lua`:

vim.o.shada = vim.o.shada ..',r\\\\machine\\share'
vim.o.shada = vim.o.shada .. ',rX:'               
vim.o.shada = vim.o.shada .. ',rX:\\'             
vim.o.shada = vim.o.shada .. ',rX:/'              

The resulting `:ju` shows a jumplist with any items beginning `\\machine\share`, `X:\` and `X:/` permanently removed. So the only jumplist items remaining were on the local drive, all network locations' jumplist items were removed. By the way the last one `X:/path/to/file.txt' did appear on a few list items for some reason.

Jumplist items matching the above were permanently removed, so even removing the above from `init.lua` didn't bring them back.

However doing the above fixed the long freeze when issuing the initial `:ju` or `<Ctrl-O>` command.

This does appear to work around the problem but doesn't appear to be an ideal fix:
- there might be other network or temporary locations added to the jumplist during an editing session, so we'd have to keep adding more of the above commands to remove them
- removal isn't ideal anyway, it's preferable to keep all jumplist items and just ignore them if files/paths (and drives in Windows) are inaccessible. Since the freeze only happens once, and everything works fine thereafter, is there a better way to fix?

Appreciate the help, issue and workaround identified 🙂

**Comment by zeertzjq** [comment]
Right, these jumps are only removed on writing shada, so setting 'shada' option may not be the ideal solution.
